apis/extension/reservation.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extension
+
+import (
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+)
+
+const (
+	// LabelReservationOrder controls the preference logic for Reservation.
+	// Reservation with lower order is preferred to be selected before Reservation with higher order.
+	// But if it is 0, Reservation will be selected according to the capacity score.
+	LabelReservationOrder = SchedulingDomainPrefix + "/reservation-order"
+
+	// AnnotationReservationAllocated represents the reservation allocated by the pod.
+	AnnotationReservationAllocated = SchedulingDomainPrefix + "/reservation-allocated"
+)
+
+type ReservationAllocated struct {
+	Name string    `json:"name,omitempty"`
+	UID  types.UID `json:"uid,omitempty"`
+}
+
+func GetReservationAllocated(pod *corev1.Pod) (*ReservationAllocated, error) {
+	if pod.Annotations == nil {
+		return nil, nil
+	}
+	data, ok := pod.Annotations[AnnotationReservationAllocated]
+	if !ok {
+		return nil, nil
+	}
+	reservationAllocated := &ReservationAllocated{}
+	err := json.Unmarshal([]byte(data), reservationAllocated)
+	if err != nil {
+		return nil, err
+	}
+	return reservationAllocated, nil
+}
+
+func SetReservationAllocated(pod *corev1.Pod, r *schedulingv1alpha1.Reservation) {
```

Comment:
Why no longer set the allocateOnce reservation succeeded here

Reply:
ReservationStatus is no longer set in the extension point implemented by the Reservation plugin, and the Controller is responsible for this matter. If a Pod is bound to a Reservation using AllocateOnce, the internal reservationInfo will have corresponding records.

Reply:
Why not keep setting the phase of reservation as succeeded in the plugin? The separation above makes criteria that require the scheduler to check if the allocateOnce reservation is assumed but still in the Available phase.

Reply:
ok